File: benchmarks/kotlin/b4_monte_carlo.kt
fun main() {
    val ITERATIONS = 1_000_000_000L
    var state = 42uL
    var inside = 0L
    for (i in 0 until ITERATIONS) {
        state = state * 6364136223846793005uL + 1442695040888963407uL
        val x = (state shr 33).toDouble() / (1uL shl 31).toDouble()
        state = state * 6364136223846793005uL + 1442695040888963407uL
        val y = (state shr 33).toDouble() / (1uL shl 31).toDouble()
        if (x * x + y * y <= 1.0) inside++
    }
    println("%.10f".format(4.0 * inside / ITERATIONS))
}

File: benchmarks/swift/b4_monte_carlo.swift
import Foundation
let ITERATIONS: UInt64 = 1_000_000_000
var state: UInt64 = 42
func nextDouble() -> Double {
    state = state &* 6364136223846793005 &+ 1442695040888963407
    return Double(state >> 33) / Double(UInt64(1) << 31)
}
var inside: UInt64 = 0
for _ in 0..<ITERATIONS {
    let x = nextDouble(), y = nextDouble()
    if x*x + y*y <= 1.0 { inside += 1 }
}
let pi: Double = 4.0 * Double(inside) / Double(ITERATIONS)
print(String(format: "%.10f", pi))
